contracts/HeapOrdering.sol

```diff
@@ -141,18 +141,24 @@ library HeapOrdering {
         Account memory accountToShift = getAccount(_heap, _rank);
         uint256 valueToShift = accountToShift.value;
         Account memory childToSwap;
+        Account memory rightChild;
         uint256 childRank = _rank << 1;
         // At this point, childRank (resp. childRank+1) is the rank of the left (resp. right) child.
 
         while (childRank <= size) {
-            // Compute the rank of the child with largest value.
-            if (
-                childRank < size &&
-                getAccount(_heap, childRank + 1).value > getAccount(_heap, childRank).value
-            ) childRank++;
-
             childToSwap = getAccount(_heap, childRank);
 
+            // Find the child with largest value.
+            if (childRank < size) {
+                rightChild = getAccount(_heap, childRank + 1);
+                if (rightChild.value > childToSwap.value) {
+                    unchecked {
+                        ++childRank; // This cannot overflow because childRank < size.
```

Comment:
I don't understand the comment. It cannot overflow because adding up just one at each step would not even fit inside a human's lifespan right?

Reply:
it depends on the frequency obviously but you get the idea

Reply:
We have `childRank < size` and `size <= type(uint256).max` (because `size` is of type `uint256`). So `childRank < type(uint256).max`, this means that it cannot overflow by adding one

Reply:
Previous comment about `childRank`:
> This variable is multiplied by 2 many times, so I don't think it applies safely here (even if it probably does, because childrank is related to a rank in an array). It is completely safe for variables that are only incremented, because you know you won't ever be able to reach 2^256 by only incrementing

Reply:
my bad my mistake was I forgot we double the value at each step!